src/auditor/dynamic_linkage.jl

```diff
@@ -313,31 +311,6 @@ end
 # Determine whether a library is a "default" library or not, if it is we need
 # to map it to `@rpath/$libname` on OSX or `\$ORIGIN/$libname` on Linux/FreeBSD
 is_default_lib(lib, oh) = false
```

Comment:
I honestly don't understand what's the purpose of this function.

Reply:
This method is for when `oh` is not a `MachOHandle` or an `ELFHandle`; e.g. when it's a `COFFHandle`, or some other handle type that we add in the future to `ObjectFile`

Reply:
The method yes, I understand, but not the function as a whole.  With this PR this is the only method of this function, so we can just remove it entirely